rtl/eclrtl/eclrtl.cpp

```diff
@@ -4892,6 +4894,69 @@ class CStrRegExprFindInstance : implements IStrRegExprFindInstance
         }
     }
 
+    bool nextMatch()
+    {
+        if (!matched)
+            return false;
+
+        const char *start = subs[0].second==subs[0].first?subs[0].second+1: subs[0].second;
+        if (start >= sample_end)
+        {
+            matched = false;
+            return matched;
+        }
+
+        matched = boost::regex_search(start, subs, *regEx);
+
+        return matched;
+    }
+
+    void getMatchSet(bool  & __isAllResult, size32_t & __resultBytes, void * & __result)
+    {
+        rtlRowBuilder out;
+        byte *outData;
+        size32_t outBytes = 0;
+
+        {
+            // workout size of results and allocate sufficient memory
+            boost::regex_iterator<char *> cur(sample, sample_end, *regEx);
+            boost::regex_iterator<char *> end;  // Default contructor creates an end of list marker
+
+            for (; cur != end; ++cur)
```

Comment:
My apologies:  My previous comment on this function may have been wrong...

I had assumed that the regex match had already found all the matches, so iterating through them was a cheap operation.  If that is not correct then your previous code reallocating the target as you found additional matches may be more efficient.  The same goes for the unicode implementation.